indico/types/utils.py

```diff
@@ -15,3 +16,16 @@ def _camel(match):
 
 def snake_to_cc(string: str):
     return re.sub(_snake_to_cc_re, _camel, string, 0)
+
+class Timer:
+    def __init__(self, timeout: int):
+        self.timeout = timeout
+
+    def _handler(self, signum, frame):
+        raise IndicoTimeoutError(self.timeout)
+
+    def run(self):
+        if self.timeout == 0:
+            raise IndicoTimeoutError(self.timeout)
+        signal.signal(signal.SIGALRM, self._handler)
```

Comment:
oh this is really slick! I dig it.

Reading the docs for the signal library, it looks like `signal.SIGALARM` is only available on Unix. I think we want this SDK to be usable on Windows as well-- is there a different signal you could use?

Reply:
ah yep will do!

Reply:
Looks like `signal` is just not overall a good choice as it's not Windows friendly in general. Just parsed out current timing logic into `Timer` and fixed the double time issues